File: node/examples/finx_api_example.js
/*
finx_api_example.js
 */
import FinX from "../finx_api.js";

let use_async = false;

let finx = FinX(use_async ? 'async' : 'socket');
let result;

async function run() {
    result = await finx.get_api_methods({
        callback: async(result, kwargs={}) => console.log('API METHODS:', result)
    });
    if (use_async)
        console.log('API METHODS:', result);
    let security_id = 'USQ98418AH10';
    let as_of_date = '2020-09-14';
    result = await finx.get_security_reference_data(
        security_id,
        {
            as_of_date: as_of_date,
            callback: async(result, kwargs={}) => console.log('SECURITY REFERENCE:', result)
        }
    );
    if (use_async)
        console.log('SECURITY REFERENCE:', result);
    result = await finx.get_security_analytics(
        security_id,
        {
            as_of_date: as_of_date,
            price: 100,
            callback: async(result, kwargs={}) => console.log('SECURITY ANALYTICS:', result)
        }
    );
    if (use_async)
        console.log('SECURITY ANALYTICS:', result);
    result = await finx.get_security_cash_flows(
        security_id,
        {
            as_of_date: as_of_date,
            price: 100,
            callback: async(result, kwargs={}) => console.log('SECURITY CASH FLOWS:', result)
        }
    );
    if (use_async)
        console.log('SECURITY CASH FLOWS:', result);
    else
        finx.clear_cache();
    result = await finx.batch(
        finx.get_security_reference_data,
        {
            'USQ98418AH10': {
                as_of_date: '2020-09-14',
                callback: async(result, kwargs={}) => console.log('BATCH SECURITY REFERENCE 1:', result)
            },
            '3133XXP50': {
                as_of_date: '2020-09-14',
                callback: async(result, kwargs={}) => {
                    console.log('BATCH SECURITY REFERENCE 2:', result);
                    process.exit(0);
                }
            }
        }
    );
    if (use_async)
        console.log('BATCH SECURITY REFERENCE:', result);
    return 1;
}

run().then(x => console.log(x));

File: fiteanalytics/finx_api_example.py
#! Python
"""
finx_api_example.py
"""
import json
from fiteanalytics.finx_api import FinX

# Initialize synchronous client with no arguments -
# checks environment variables for credentials
finx = FinX()

# Get API methods
print('\n*********** API Methods ***********')
api_methods = finx.get_api_methods()
print(json.dumps(api_methods, indent=4))

security_id = 'USQ98418AH10'
as_of_date = '2020-09-14'

# Get security reference data
print('\n*********** Security Reference Data ***********')
reference_data = finx.get_security_reference_data(
    security_id,
    as_of_date=as_of_date)
print(json.dumps(reference_data, indent=4))

# Get security analytics
print('\n*********** Security Analytics ***********')
analytics = finx.get_security_analytics(
    security_id,
    as_of_date=as_of_date,
    price=100)
print(json.dumps(analytics, indent=4))

# Get projected cash flows
print('\n*********** Security Cash Flows ***********')
cash_flows = finx.get_security_cash_flows(
    security_id,
    as_of_date=as_of_date,
    price=100)
print(json.dumps(cash_flows, indent=4))
